inspirehep/dojson/common/base.py

```diff
@@ -318,3 +322,70 @@ def deleted_records2marc(self, key, value):
     return {
         'a': inspire_dojson_utils.get_recid_from_ref(value)
     }
+
+
+@conferences.over('urls', '^8564')
+@experiments.over('urls', '^8564')
+@hep.over('urls', '^856.[10_28]')
+@hepnames.over('urls', '^856.[10_28]')
+@institutions.over('urls', '^856.[10_28]')
+@jobs.over('urls', '^856.[10_28]')
+@journals.over('urls', '^856.[10_28]')
+def urls(self, key, value):
+    """URL to external resource."""
+    urls = []
+    value = utils.force_list(value)
+    for val in value:
+        url_dict = {}
+        description = utils.force_list(val.get('y'))
+        url_values = utils.force_list(val.get('u'))
+        for u in url_values:
+            url_dict = {}
+            url_dict['value'] = u
+            if description:
+                url_dict['description'] = description[0]
+            urls.append(url_dict)
+
+    return inspire_dojson_utils.remove_duplicates_from_list_of_dicts(urls)
+
+
+@hepnames.over('field_categories', '^65017')
+@conferences.over('field_categories', '^65017')
+@institutions.over('field_categories', '^65017')
+@jobs.over('field_categories', '^65017')
+@experiments.over('field_categories', '^65017')
+@hep.over('field_categories', '^650[1_][_7]')
+def field_categories(self, key, value):
+    """Field categories."""
+    value = utils.force_list(value)
+    field_categories = self.get('field_categories', [])
+    for element in value:
+        _scheme = element.get('2')
+        if isinstance(_scheme, (list, tuple)):
+            _scheme = _scheme[0]
+            # FIXME: raise Exception("Multiple schemes")
+        source = element.get('9')
+
+        if 'a' in element:
+            el = utils.force_list(element['a'])
+            for val in el:
+                field_dict = {}
+                corrected_val = classify_field(val)
+                if corrected_val:
+                    field_dict['term'] = corrected_val
+                    field_dict['_term'] = val
+                    field_dict['scheme'] = 'INSPIRE'
+                    if _scheme:
+                        field_dict['_scheme'] = _scheme
+                        # FIXME: right now '_scheme' is what it says in the
+                        # MARCXML. Should the original term's real scheme be
+                        # detected and put here instead?
+                    if source:
+                        if 'automatically' in source:
+                            source = 'INSPIRE'
+                        field_dict['source'] = source
+                    field_categories.append(field_dict)
+                # else:
+                    # raise Exception("Field value type should be str or unicode")
```

Comment:
Not sure what's happening here, since we are guarding against `corrected_val` being a falsy value...